mobile/src/home-view.js

```diff
@@ -0,0 +1,278 @@
+    questionScore = questions.map(function(x) { return Object.assign({id: x.key, score: 0}) })
+
+    this.findScore(questionScore, votes, questions)
+    if (showRecent === true) {
+    }
+  
+ 
+   
+    return (
+      <KeyboardAvoidingView style={s.container} behavior={Platform.select({ios: "padding", android: null})}>
+        <TitleBar title="Keynote" client={client} signin={this.signin} />
+        <View style={s.compose}>
+          <TextInput style={s.composeText} placeholder="Add Question..."
+            value={this.state.question}
+            onChangeText={question => this.setState({question})} 
+            maxLength={250}
+            />
+          <View style={s.sendButtons}>
+            <TouchableOpacity style={s.sendButton} onPress={this.createSharedTask}><Text style={s.sendButtonText}>Submit 📢</Text></TouchableOpacity>
+          </View>
+        </View>
+        <View style={s.counter}>
+        <Text>{this.state.question.length}/250 </Text>
+        </View>
+        
+
+        <ScrollView style={s.scroll}>
+        
+          {questions.map(question => (
+            <View key={question.id} style={s.task}>
+              {/* { renderCreator(question) } */}
+              <Text style={s.taskText}>{question.text}</Text>
+              <TouchableOpacity onPress={() => this.newVote(question)}><Text style={s.checkmark}>👍 </Text></TouchableOpacity>
+            </View>
+          ))} 
+        </ScrollView>
+        
+       
+      </KeyboardAvoidingView>
+    )
+  }
+ 
+
+
+  findScore(questionScore, votes, questions){
+    votes.forEach(function(a) {
+    questionScore.find(function(value, index) {
+      if (value.id === a.question) {
+        value.score += a.vote
+      }
+    })
+  });
+  this.findOrder(questionScore, questions)
+  }
+
+
+  findOrder(questionScore, questions){
+  
+    questionScore.forEach(function(a){
+      questions.find(function(value, index){
+        console.log(value.creator)
+        if (value.key === a.id) {
+          value.score = a.score
+        }  
+      })
+    });
+     questions.sort(function (a,b){
+      return b.score - a.score
+    })
+  }
+
+  findOrderDate(questions){
+
+    //need to sort by date created
+
+  }
+
+  createSharedTask = () => this.createQuestion(fbc.database.public.allRef)
+ 
+  createQuestion(ref) {
+    console.log(this.state.question)
+    if (this.user && this.state.question) {
+      var time = new Date().getTime()
+      ref('questions').push({
+        text: this.state.question,
+        creator: client.currentUser.firstName + " " + client.currentUser.lastName,
+        score : 0,
+        dateCreate: time
+      })
+      .then(() => this.setState({question: ''}))
+      .catch (x => console.error(x))
+    }    
+  }
+
+  createVote(voteNum, question){
+    ref = fbc.database.public.allRef
+    if (this.user && question) {
+      ref('votes').push({
+        question: question.key,
+        creator: client.currentUser.id,
```

Comment:
Instead of having client-side logic to check whether the current user has voted for a particular question before adding the vote, you could simply use the `currentUser.id` as the key under `votes`.  So while `push(...)` will auto-generate a key for that vote (e.g. `{ abc: {question: "qx", creator: "uy", vote: 1}, def: {question: "qx", creator: "uz", vote: 1} }`), you could simply use some nested layers in the DB, e.g. `fbc.database.public.allRef('votes').child(question.key).child(client.currentUser.id).set(1)`
resulting in votes in the DB looking like: { qx: {uy: 1, uz: 1} }

This has a couple advantages:
1. Voting again simply overwrites the old vote.
2. Rather than filtering across all votes for all questions, you can set up `child_*` change handlers for each question (e.g. `votes/qx`), and know that any votes that come in are only for question `qx`.  So if a `child_added` for `votes/qx` appears, you can just add that vote to the sum total for that question, rather than recomputing all votes across all questions.